io/system/system.go

```diff
@@ -59,24 +59,37 @@ type StageEvent struct {
 type Stage uint8
 
 const (
-	// StagePaused is the Stage for inactive Windows.
-	// Inactive Windows don't receive FrameEvents.
+	// StagePaused is the stage for windows that have no on-screen representation.
+	// Paused windows don't receive FrameEvent.
 	StagePaused Stage = iota
-	// StateRunning is for active Windows.
+	// StageInactive is the stage for windows that are visible, but not active.
+	// Inactive windows receive FrameEvent.
+	StageInactive
+	// StageRunning is for active and visible Windows.
+	// Running windows receive FrameEvent.
 	StageRunning
 )
 
+// String implements fmt.Stringer.
 func (l Stage) String() string {
 	switch l {
 	case StagePaused:
 		return "StagePaused"
+	case StageInactive:
+		return "StageInactive"
 	case StageRunning:
 		return "StageRunning"
 	default:
 		panic("unexpected Stage value")
 	}
 }
 
+// IsReceivingFrames returns true if the window is receiving
+// FrameEvent.
+func (l Stage) IsReceivingFrames() bool {
+	return l >= StageInactive
+}
+
```

Comment:
What's the use case(s) for exporting IsReceivingFrames? It seems to me it shouldn't be exported.

Reply:
I think the use-case is to replace any `if stage < system.StageRunning`. If new Stage is added in the future, the `IsReceivingFrames()` can be updated internally, without  need to change every `if stage < system.StageRunning`:

https://github.com/gioui/gio/blob/a1c783812504da29db3df7489e453d7c8bb62e7d/app/window.go#L426-L431

It can't be private, but that lives on `io/system` and it's used by `app`. I can remove it and replace everything to `if stage != system.StagePaused` or  `stage >= system.StageInactive`, I just thought that it will be harder to understand and maintain.

Reply:
Adding new stages is a major addition and code that uses the stage should be inspected (for possible updating). In other words, we haven't added new stages for a long time (years?) and I don't expect we will for a long time (if ever).